src/javascript/app/Stores/Modules/Trading/trade-store.js

```diff
@@ -418,6 +419,16 @@ export default class TradeStore extends BaseStore {
                 proposal_info      : {},
             });
 
+            // To prevent infinite loop when changing from advanced end_time to digit type contract
+            if (obj_new_values.contract_type && this.root_store.ui.is_advanced_duration) {
+                if (isDigitTradeType(obj_new_values.contract_type)) {
```

Comment:
Is it possible to flatten the nested if statement here?

Reply:
Sure, will do in other PR because testing has already finished for this. Don't want to risk the changes for now. 😅